src/components/credentials/credentialsContext.js

```diff
@@ -0,0 +1,292 @@
+import React, { useState } from 'react';
+import { useShallowCompareEffect } from 'react-use';
+import { AlertVariant, List, ListItem } from '@patternfly/react-core';
+import { ContextIcon, ContextIconVariant } from '../contextIcon/contextIcon';
+import { reduxActions, reduxTypes, storeHooks } from '../../redux';
+import { apiTypes } from '../../constants/apiConstants';
+import { helpers } from '../../common';
+import { translate } from '../i18n/i18n';
+import { useConfirmation } from '../../hooks/useConfirmation';
+
+/**
+ * On expand a row facet.
+ *
+ * @param {object} options
+ * @param {Function} options.useDispatch
+ * @returns {Function}
+ */
+const useOnExpand = ({ useDispatch: useAliasDispatch = storeHooks.reactRedux.useDispatch } = {}) => {
+  const dispatch = useAliasDispatch();
+
+  return ({ isExpanded, cellIndex, data }) => {
+    dispatch({
+      type: isExpanded ? reduxTypes.credentials.EXPANDED_CREDENTIAL : reduxTypes.credentials.NOT_EXPANDED_CREDENTIAL,
+      viewType: reduxTypes.view.CREDENTIALS_VIEW,
+      item: data.item,
+      cellIndex
+    });
+  };
+};
+
+/**
+ * On refresh view.
+ *
+ * @param {object} options
+ * @param {Function} options.useDispatch
+ * @returns {Function}
+ */
+const useOnRefresh = ({ useDispatch: useAliasDispatch = storeHooks.reactRedux.useDispatch } = {}) => {
+  const dispatch = useAliasDispatch();
+
+  return () => {
+    dispatch({
+      type: reduxTypes.credentials.UPDATE_CREDENTIALS
+    });
+  };
+};
+
+/**
+ * Credential actions, delete and edit.
+ *
+ * @param {object} options
+ * @param {Function} options.deleteCredentials
+ * @param {Function} options.t
+ * @param {Function} options.useConfirmation
+ * @param {Function} options.useDispatch
+ * @param {Function} options.useSelectorsResponse
+ * @returns {{onDelete: Function, onEdit: Function}}
+ */
+const useOnCredentialAction = ({
```

Comment:
optional:
useOnCredentialAction should probably be broken out into 2 hooks since the majority of this hook is used for `onDelete`, it would help remove confusion around maintenance